enterprise/cli/provisionerkeys.go

```diff
@@ -144,6 +145,29 @@ func (r *RootCmd) provisionerKeysList() *serpent.Command {
 	return cmd
 }
 
+type provisionerKeyListItem struct {
+	CreatedAt time.Time `json:"created_at" table:"created at" format:"date-time"`
+	Name      string    `json:"name" table:"name,default_sort"`
+	Tags      string    `json:"tags" table:"tags"`
+}
+
+func provisionerKeyList(items []codersdk.ProvisionerKey) []provisionerKeyListItem {
+	var out []provisionerKeyListItem
+	for _, item := range items {
+		tags := []string{}
+		for k, v := range item.Tags {
+			tags = append(tags, fmt.Sprintf("%s=%s", k, v))
+		}
+		out = append(out, provisionerKeyListItem{
+			Name:      item.Name,
+			CreatedAt: item.CreatedAt,
+			Tags:      strings.Join(tags, ", "),
+		})
+	}
+
+	return out
+}
```

Comment:
We should use the original keys, just change the default columns.

```golang
cliui.TableFormat([]codersdk.ProvisionerKey{}, []string{"Created At", "Name", "Tags"}),
```

Then you can optionally grab the other columns if you'd like with `-c`

Reply:
I need to convert tags render from `map[my:val,your:val]` to `my=val, your=val` which is why I made the new struct.

Reply:
Ah. IIRC we call `.String()` if present on the type.

We could make a type, and implement `String()` on it.

```golang
type ProvisionerKeyTags map[string]string

func (p ProvisionerKeyTags) String() string {
	return "yay"
}

type ProvisionerKey struct {
	ID             uuid.UUID         `json:"id" table:"-" format:"uuid"`
	CreatedAt      time.Time         `json:"created_at" table:"created at" format:"date-time"`
	OrganizationID uuid.UUID         `json:"organization" table:"organization id" format:"uuid"`
	Name           string            `json:"name" table:"name,default_sort"`
	Tags           ProvisionerKeyTags `json:"tags" table:"tags"`
	// HashedSecret - never include the access token in the API response
}
```

I prefer not to make type duplicates for formatting where we can avoid it.